### ISSUE TITLE
Checking for application updates always shows "Up to date" even though a new version is actually available

### ISSUE THREAD

**Issue by atanasuzunov** [microsoft/XmlNotepad#178]
It looks the new version checker does not work as expected. It always shows "Up to date" even though there is a new version available. 
The behavior is tested on 2.8.0.52 when there is 2.8.0.54 available. The corresponding 2.8.0.54 version entry is present in the https://github.com/microsoft/XmlNotepad/blob/master/src/Updates/Updates.xml file.

**Comment by lovettchris** [comment]
Thanks for the bug report, can you specify which installer you used to install XML notepad?
1. clickonce
2. standalone .msi installer
3. winget

**Comment by atanasuzunov** [comment]
I'm using 2 - the standalone one.

**Comment by lovettchris** [comment]
Got it - so yes, the standalone installer does not "auto-update".  You will have to download [the new zip](https://github.com/microsoft/XmlNotepad/releases/tag/2.8.0.54) and install it manually.  If you want auto-updating you need to use the [ClickOnce installer](https://microsoft.github.io/XmlNotepad/#install/).

**Comment by atanasuzunov** [comment]
I'm aware that the program does not auto update. I always install the new versions manually. The problem is that the notifier does not show that a new version is available - it always shows "Up to date" when performing the check for new versions, even when there is really a new one available. 
This functionality is broken since one of the latest updates - not sure which one exactly. It worked fine before that.

**Comment by lovettchris** [comment]
Ah, sorry I misunderstood.  The check for update actually pings the <UpdateLocation> specified in your `"%LOCALAPPDATA%\Microsoft\Xml Notepad\XmlNotepad.settings"`.  So please check your settings contains this line:

```xml
<UpdateLocation>https://lovettsoftwarestorage.blob.core.windows.net/downloads/XmlNotepad/Updates.xml</UpdateLocation>
```
as a child of `<Settings>`.

**Comment by lovettchris** [comment]
Oh, and there's one more thing, there is a cached "LastUpdateCheck" so that the app does not check more than once per "UpdateFrequency" which defaults to once per day:
```xml
<LastUpdateCheck>3/14/2022 12:33 AM</LastUpdateCheck>
<UpdateFrequency>1.00:00:00</UpdateFrequency>
```
so when I have these values in my settings file when I launch version 2.8.0.53, I do not see the green update available button.  But I will see it tomorrow.

**Comment by lovettchris** [comment]
Doh, did some further debugging and found a bug.  A version check was reversed.  Pushed a fix https://github.com/microsoft/XmlNotepad/commit/5203b1572f69ee892e562c2acd5ac911040d3f82 just now, the fix will be included in the next release.

**Comment by atanasuzunov** [comment]
Thanks for fixing it! 
The "%LOCALAPPDATA%\Microsoft\Xml Notepad\XmlNotepad.settings" on my side looks fine, so I guess your fix will solve the problem in the next release.

**Comment by lovettchris** [comment]
Fixed in 2.8.0.55

**Comment by Pierre-de-la-Verre** [comment]
The question is: How to get the users of versions 52 or 53 out of the dead-end "There are no updates.."?

**Comment by lovettchris** [comment]
If they used ClickOnce there is no problem, it still updates automatically.  If they used standalone installer they will have to re-install manually, but yeah, they've have to think to do it, they won't get a reminder.  Same for winget.

**Comment by lovettchris** [comment]
Should be fixed in 2.8.0.56

**Comment by atanasuzunov** [comment]
It is already fixed in 2.8.0.55.

PS: 2.8.0.56 gave me some file missing exception at app startup, so I reverted to 2.8.0.55.
